src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java

```diff
@@ -825,6 +646,10 @@ private String key() {
                 return TaskListener.NULL;
             }
         }
+        @Override public InputStream getLog() throws IOException {
```

Comment:
Tons of Jenkins APIs expect a `TaskListener` or `OutputStream` for output, and expect an `AnnotatedLargeText` or `InputStream` for input, so streams are mandated at some level I think. Of course a logging implementation might internally work with lines.